src/live_restore/live_restore_fs.c

```diff
@@ -886,17 +885,28 @@ __wti_live_restore_fs_fill_holes(WT_FILE_HANDLE *fh, WT_SESSION *wt_session)
     WT_TIMER timer;
     uint64_t msg_count = 0;
     bool finished = false;
+    char *buf = NULL;
+    WT_SESSION_IMPL *session = (WT_SESSION_IMPL *)wt_session;
+    WT_RET(
+      __wt_malloc(session, ((WTI_LIVE_RESTORE_FS *)S2C(session)->file_system)->read_size, &buf));
 
-    __wt_timer_start((WT_SESSION_IMPL *)wt_session, &timer);
+    __wt_timer_start(session, &timer);
     while (!finished) {
-        WTI_WITH_LIVE_RESTORE_EXTENT_LIST_WRITE_LOCK((WT_SESSION_IMPL *)wt_session,
-          (WTI_LIVE_RESTORE_FILE_HANDLE *)fh,
-          ret = __live_restore_fill_hole(fh, wt_session, &timer, &msg_count, &finished));
-        WT_RET(ret);
-        WT_RET(WT_SESSION_CHECK_PANIC(wt_session));
+        WTI_WITH_LIVE_RESTORE_EXTENT_LIST_WRITE_LOCK(session, (WTI_LIVE_RESTORE_FILE_HANDLE *)fh,
+          ret = __live_restore_fill_hole(fh, wt_session, buf, &timer, &msg_count, &finished));
+        WT_ERR(ret);
+
+        /*
+         * Because this loop can run for a very long time, ensure the system has not entered a panic
+         * state in the meantime.
+         */
+        WT_ERR(WT_SESSION_CHECK_PANIC(wt_session));
     }
 
-    return (0);
+err:
+    __wt_free(session, buf);
```

Comment:
This function still needs a sync call added, I think.

Reply:
042d20ceb